src/system/game-data.ts

```diff
@@ -795,44 +791,50 @@ export class GameData {
     return true; // TODO: is `true` the correct return value?
   }
 
-  public saveTutorialFlag(tutorial: Tutorial, flag: boolean): boolean {
+  /**
+   * Retrieve the tutorial seen flags from local storage
+   * @returns the flags saved in local storage if they exist, otherwise an empty array
+   */
+  private getTutorialFlags(): (boolean | null)[] {
     const key = getDataTypeKey(GameDataType.TUTORIALS);
-    let tutorials: object = {};
-    if (localStorage.hasOwnProperty(key)) {
-      tutorials = JSON.parse(localStorage.getItem(key)!); // TODO: is this bang correct?
-    }
+    let tutorials = [];
 
-    Object.keys(Tutorial).forEach((key) => {
-      if (key === Tutorial[tutorial]) {
-        tutorials[key] = flag;
-      } else {
-        tutorials[key] ??= false;
+    const lsItem = localStorage.getItem(key);
+    if (lsItem) {
+      try {
+        tutorials = JSON.parse(lsItem);
+      } catch (err) {
+        console.warn("Failed to parse tutorial data from local storage", err);
       }
-    });
-
-    localStorage.setItem(key, JSON.stringify(tutorials));
-
-    return true;
+    }
+    return tutorials;
   }
 
-  public getTutorialFlags(): TutorialFlags {
+  /**
+   * Registers the given tutorial as seen in local storage
+   * @param tutorial the {@linkcode Tutorial} to update the flag for
+   * @returns `true` if saving was successful, `false` otherwise
+   */
+  public saveTutorialAsSeen(tutorial: Tutorial): boolean {
     const key = getDataTypeKey(GameDataType.TUTORIALS);
-    const ret: TutorialFlags = {};
-    Object.values(Tutorial)
-      .map((tutorial) => tutorial as Tutorial)
-      .forEach((tutorial) => (ret[Tutorial[tutorial]] = false));
-
-    if (!localStorage.hasOwnProperty(key)) {
-      return ret;
-    }
-
-    const tutorials = JSON.parse(localStorage.getItem(key)!); // TODO: is this bang correct?
-
-    for (const tutorial of Object.keys(tutorials)) {
-      ret[tutorial] = tutorials[tutorial];
+    const tutorials = this.getTutorialFlags();
+    tutorials[tutorial] = true;
+    try {
+      localStorage.setItem(key, JSON.stringify(tutorials));
+      return true;
+    } catch (err) {
+      console.error("Failed to saved tutorial data in local storage", err);
+      return false;
     }
+  }
 
-    return ret;
+  /**
+   * Checks if the given tutorial is marked as seen in local storage
+   * @param tutorial the {@linkcode Tutorial} to get the flag for
+   * @returns `true` if the tutorial has already been seen, `false` otherwise
+   */
+  public wasTutorialSeen(tutorial: Tutorial): boolean {
```

Comment:
NIT: using “was” is an uncommon appearance for me

```suggestion
  public isSeenTutorial(tutorial: Tutorial): boolean {
```

Reply:
LS key updated to seenTutorials to match the seenDialogues one